### PR TITLE
Improve logging

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Summary of Changes

- Replace `LogDebug` with `LogInformation`
- In DynamoDbGateway, move load and save logging into private methods
- In DynamoDbGateway, add logging when calling each method
- Add logging to EditTenureDetailsUseCase

### CODE REVIEW

File: TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs [hunk 1]
```diff
@@ -93,7 +93,6 @@ public async Task GetEntityByIdReturnsNullIfEntityDoesntExist()
             var response = await _classUnderTest.GetEntityById(request).ConfigureAwait(false);
 
             response.Should().BeNull();
-            _logger.VerifyExact(LogLevel.Debug, $"Calling IDynamoDBContext.LoadAsync for id {request.Id}", Times.Once());
```
[the remark this answers is not in the document]
Comment (LBHCallumM) [reply TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:96]: In my opinion, logging is an implementation detail. These tests, based on the name, shouldn't have broken when I modified the logging. I don't see them as adding any value.

[the remark this answers is not in the document]
Comment (Asura5) [reply TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:96]: That's interesting - have you considered the logging output in the wider context? What about if a cloudwatch alarm is based of a certain log being emitted? You'd want to check that log is always produced with a test

Try not to fall in to the trap of "I don't see it as important, so therefore, it's not important"

[the remark this answers is not in the document]
Comment (LBHCallumM) [reply TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:96]: You're right. I did make an assumption.

However, in these specific use cases, there is no useful information being logged out that could be used by a cloudwatch alarm (I have also checked, there are no alarms). Secondly, the logs were not appearing in Cloudwatch in the first place, due to the sampling rate. 

But you have highlighted a blind spot I have. I shouldn't assume something has been implemented in the same way that I would have approached it.

Would you rather stick to a policy of not unnecessarily modifying anything that's tested for in any unit tests? Or simply that I should be more cautious when doing so?

[the remark this answers is not in the document]
Comment (Asura5) [reply TenureInformationApi.Tests/V1/Gateways/DynamoDbGatewayTests.cs:96]: It's up to you - I will generally challenge anywhere a developer has removed testing detail, without elaborating on why. So carry on as you are but expect to continue to have to justify your actions (which you have done perfectly in this case)

**Review by Asura5 — APPROVED** [review]

**Pull request merged**